examples/e14_global_data/src/main.rs

```diff
@@ -0,0 +1,215 @@
+    }
+}
+
+#[tokio::main]
+async fn main() {
+    let token = env::var("DISCORD_TOKEN").expect("Expected a token in the environment");
+    
+    let framework = StandardFramework::new()
+        .configure(|c| c
+            .with_whitespace(true)
+            .prefix("~")
+        )
+        .before(before)
+        .group(&GENERAL_GROUP);
+
+    let mut client = Client::builder(&token)
+        .event_handler(Handler)
+        .framework(framework)
+        .await
+        .expect("Err creating client");
+
+    // This is wherere we can initially insert the data we desire into the "global" data TypeMap.
+    // client.data is wrapped on a RwLock, and since we want to insert to it, we have to open it in
+    // write mode, but there's a small thing catch:
+    // There can only be a single writer to a given lock open in the entire application, this means
+    // you can't open a new write lock until the previous write lock has closed.
+    // This is not the case with Read locks, read locks can be open indefinitely, BUT as soon as
+    // you need to open the lock in write mode, all the read locks must be closed.
+    //
+    // You can find more information about deadlocks in the Rust Book, ch16-03:
+    // https://doc.rust-lang.org/book/ch16-03-shared-state.html
+    //
+    // All of this means that we have to keep locks open for the least time possible, so we put
+    // them inside a block, so they get closed automatically when droped.
+    // If we don't do this, we would never be able to open the data lock anywhere else.
+    {
+        // Open the data lock in write mode, so keys can be inserted to it.
+        let mut data = client.data.write().await;
+
+        // The CommandCounter Value has the following type:
+        // Arc<RwLock<HashMap<String, u64>>>
+        // So, we have to insert the same type to it.
+        data.insert::<CommandCounter>(Arc::new(RwLock::new(HashMap::default())));
+
+        data.insert::<MessageCount>(Arc::new(AtomicUsize::new(0)));
+    }
+
+    if let Err(why) = client.start().await {
+        eprintln!("Client error: {:?}", why);
+    }
+}
+
+#[command]
+async fn ping(ctx: &Context, msg: &Message) -> CommandResult {
+    msg.reply(ctx, "Pong!").await?;
+
+    Ok(())
+}
+
+/// Usage: `~command_usage <command_name>`
+/// Example: `~command_usage ping`
+#[command]
+async fn command_usage(ctx: &Context, msg: &Message, mut args: Args) -> CommandResult {
+    let command_name = match args.single_quoted::<String>() {
+        Ok(x) => x,
+        Err(_) => {
+            msg.reply(ctx, "I require an argument to run this command.").await?;
+            return Ok(());
+        }
+    };
+
+    // Yet again, we want to keep the locks open for the least time possible.
+    let amount = {
+        // Since we only want to read the data and not write to it, we open it in read mode;
+        // and since this is open in read mode, it means that there can be multiple locks open at
```

Comment:
Here I suggest two ways:
1) Remove the conjunction *and*.
```suggestion
        // Since we only want to read the data and not write to it, we open it in read mode;
        // since this is open in read mode, it means that there can be multiple locks open at
```
The semicolon already does the job of joining two independent clauses together, removing the need for conjunctions such as *and*, *but*, *for*, etc..

2) Change the semicolon to a comma.
```suggestion
        // Since we only want to read the data and not write to it, we open it in read mode,
        // and since this is open in read mode, it means that there can be multiple locks open at
```